### PR TITLE
Correct the sorting of the current edition

### PR DESCRIPTION
**Pull request by matthias-ronge**
Fixes the sorting of multiple issues on one day in the `year.xml` file for the current issue of the day, deriving the sort order from the meta-data entry `CurrentNoSorting`.

### CODE REVIEW

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java [hunk 1]
```diff
@@ -771,6 +771,48 @@ private static Integer positionByRank(List<DocStruct> siblings, String metadataT
...
+     * {@link DocStruct#addChild(Integer, DocStruct)} to simply append the new
+     * child at the end.
+     *
+     * @param siblings
+     *            brothers and sisters of the child to add
+     * @param rank
+     *            rank of the child to insert
+     * @return the index position to insert the child
+     */
+    private static Integer positionByRank(List<DocStruct> siblings, Integer rank) {
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: Why did you add this method as a static one?

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java [hunk 2]
```diff
@@ -771,6 +771,48 @@ private static Integer positionByRank(List<DocStruct> siblings, String metadataT
...
+        if (siblings != null || rank != null) {
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: Is your condition `||` correct? If `siblings` is `null` and `rank` not `null` then it would create a null pointer exception in the next line.

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java [hunk 3]
```diff
@@ -771,6 +771,48 @@ private static Integer positionByRank(List<DocStruct> siblings, String metadataT
...
+            SIBLINGS: for (int i = 0; i < siblings.size(); i++) {
+                DocStruct aforeborn = siblings.get(i);
```
Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: As I did not find any translation: what did `aforeborn` mean or is there some spelling error?

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java (hunk 1 above)
Comment (matthias-ronge) [reply Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: It doesn’t need access to fields of the class instance. I basically declare any method static if it is. You do not have to, I know. But why should I leave the assumption of a dependency if I know that there is no such?

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java (hunk 3 above)
Comment (matthias-ronge) [reply Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: children − _Kinder_; siblings − _Geschwister_; aforeborn − _Vorgeborene/r, älterer Bruder/ältere Schwester_
At least now I wonder: Why are the outgoing relations of a tree structure actually called children?

File: Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java (hunk 2 above)
Comment (matthias-ronge) [reply Goobi/src/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java]: Yes, that was wrong, of course. Thank you. Of course it should be called `&&`.

**Comment by henning-gerhardt** [comment]
@matthias-ronge: We tested your change but the ordering on the same day is still wrong. Current issue will be first regardless of its real position.

**Comment by henning-gerhardt** [comment]
Sorry, I deployed the wrong branch without your changes to our test system. We are testing now a version with your changes.

**Comment by matthias-ronge** [comment]
@henning-gerhardt, can this be merged?

**Comment by henning-gerhardt** [comment]
First test exports looks good, so it could be merged in couple of next days.
